pkgs/README.md

```diff
@@ -384,7 +384,7 @@ All versions of a package _must_ be included in `all-packages.nix` to make sure
 * `meta.license` must be set and match the upstream license.
   * If there is no upstream license, `meta.license` should default to `lib.licenses.unfree`.
   * If in doubt, try to contact the upstream developers for clarification.
-* `meta.mainProgram` must be set when appropriate.
+* `meta.mainProgram` must be set when there is any main binary for the package in `$out/bin`. Do not set this to `pname`; use a hardcoded string.
```

Comment:
Maybe it's worth mentioning examples eg a package which is a collection of binaries where there is no obvious 'main' binary

Reply:
What examples do you have in mind? Apparently `coreutils` isn't the best counterexample based on the linked issue.

Reply:
did my reply in linked issue made `coreutils` invalid counterexample ?

coreutils has bin/coreutils, maybe it's good to have mainProgram to point to it. One other package I thought of is imagemagick, and it has `bin/magic` thats essentially same as `bin/coreutils` ( it's a proxy to execute other tools )

can people think of other examples ?

Reply:
imagemagick's `magick` is a different case, actually they recommend using `magick` directly, and the old binaries are only there for backwards compat

Reply:
good to know.

do others agree to that coreutils mainProgram should be "coreutils" ? how about blackbox ? that doesn't seem to have an obvious candidate for mainProgram ?

Reply:
sorry, I'll try to simplify the question